### PR TITLE
[6153] Cut & Paste content does not carry over the content's history

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6153

### CODE REVIEW

File: static-assets/components/cstudio-common/common-api.js [hunk 1]
```diff
@@ -3691,7 +3691,7 @@ var nodeOpen = false,
 
         this.getVersionHistory(site, contentTO, {
           success: function (response) {
-            callback.success(response.versions[0].versionNumber);
+            callback.success(response.items[0].versionNumber);
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Didn't we make the service return directly the versions array?

File: ui/app/src/models/Version.ts [hunk 2]
```diff
@@ -14,21 +14,20 @@
...
+import { DetailedItem } from './Item';
 import ApiResponse from './ApiResponse';
 import ContentInstance from './ContentInstance';
+import Person from './Person';
 
-export interface LegacyVersion {
-  lastModifiedDate: string;
-  lastModifier: string;
-  versionNumber: string;
+export interface ItemVersion {
```
Comment (rart) [inline ui/app/src/models/Version.ts]:
```suggestion
export interface ItemHistoryEntry {
```

**Review by rart — CHANGES_REQUESTED** [review]
Please update CHANGELOG with important & breaking changes (for example the change in return of the service functions)

File: ui/app/src/CHANGELOG.md [hunk 3]
```diff
@@ -5,3 +5,5 @@
   * Add `submitter` prop of type `Person`.
   * Add `dateSubmitted` prop of type string.
 * Update PendingApprovalDashlet and ScheduledDashlet to use new `submitter` and `dateSubmitted` props.
+* Update return type of configuration/fetchHistory and content/fetchItemHistory to `ItemHistoryEntry[]`.
+* Update content/fetchItemHistory service to use new `/studio/api/2/content/item_history` API.
```
Comment (rart) [inline ui/app/src/CHANGELOG.md]: This goes over 4.1.2, dude...

File: ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx [hunk 4]
```diff
@@ -168,7 +162,7 @@ export function HistoryDialogContainer(props: HistoryDialogContainerProps) {
         ])
       );
     } else if (isItemPreviewable) {
-      fetchContentByCommitId(site, item.path, version.versionNumber).subscribe((content) => {
+      fetchContentByCommitId(site, ensureSingleSlash(`/${versionPath}`), version.versionNumber).subscribe((content) => {
```
Comment (rart) [inline ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx]: Why would a path not have it's starting slash?
Why is this needed only on this code path?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx (hunk 4 above)
Comment (jvega190) [reply ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx]: paths are coming without starting slash in the item history API, but only `fetchContentByCommitId` seems to be failing when the path doesn't have its starting slash

Comment (rart) [reply ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx]: @sumerjabri @jmendeza is it expected that the paths come without their leading slash [or is it a bug]?

Comment (jmendeza) [reply ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx]: @rart I would say it is a bug, since everywhere we used the path with the leading slash.

Comment (jmendeza) [reply ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx]: Fixed here https://github.com/craftercms/studio/pull/3314

Comment (rart) [reply ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx]: @jvega190 please update and adjust

File: ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx [hunk 5]
```diff
@@ -51,14 +50,16 @@ import {
...
 import SingleItemSelector from '../SingleItemSelector';
-import { SuspenseWithEmptyState } from '../Suspencified/Suspencified';
 import VersionList from '../VersionList';
 import DialogFooter from '../DialogFooter/DialogFooter';
 import { HistoryDialogPagination } from './HistoryDialogPagination';
 import { historyStyles } from './HistoryDialog';
 import useSelection from '../../hooks/useSelection';
 import useFetchSandboxItems from '../../hooks/useFetchSandboxItems';
 import { UNDEFINED } from '../../utils/constants';
+import { ensureSingleSlash } from '../../utils/string';
```
Comment (rart) [inline ui/app/src/components/HistoryDialog/HistoryDialogContainer.tsx]: Is this still in use? Don't see it used on the diff insertions and seems like a new dependency. You're probably getting warnings on the build?

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]
